specs/span-batches.md

```diff
@@ -153,6 +154,20 @@ decoding. For example, lets say bad batcher wrote span batch which `block_count
 the explicit limit, not trying to consume data until EOF is reached. We can also safely preallocate memory for decoding
 because we know the upper limit of memory usage.
 
+## Span batch Hard Fork Rule
+
+Span batch hard fork is activated based on timestamp.
+
+Activation Rule: `x != null && x >= upgradeTime && y != null && y >= upgradeTime`
+
+Let `inclusion_block` be the L1 block when the span batch was first fully derived.
```

Comment:
Is "fully derived" the right term here? The derivation spec talks about the channel being "ready" (https://github.com/ethereum-optimism/optimism/blob/2f523fbb9315a6ea72a971535dd1632806a9e282/specs/derivation.md#reading).  Though I suspect we should be doing this validation based on the _first_ L1 block the span batch was included in and not the last. That would be the `channel.open_l1_block`  from https://github.com/ethereum-optimism/optimism/blob/2f523fbb9315a6ea72a971535dd1632806a9e282/specs/derivation.md#timeouts

Reply:
The definition of `inclusion_block` is based on the [derivation spec#batch-queue](https://github.com/ethereum-optimism/optimism/blame/96a24cc3f6f5bbbb8f42011988208619a41344fd/specs/derivation.md#L589C5-L589C5),

> `inclusion_block_number` is the L1 block number when `batch` was first _fully derived_, i.e. decoded and output by the previous stage.

so I think `fully derived` is the right term. Span batch docs is also written based on the term `inclusion_block`.

Reply:
I like the new proposal, and thank you for the diagram, it makes it very clear.

Reply:
Yeah I like using the origin of the first L2 block too. It should fit very naturally into the batcher's logic for activating span batches too.